components/data-oriented-programming/code/java-bbe/pattern-matching.md

```diff
@@ -1,6 +1,6 @@
 ---
 title: 'Pattern matching'
-description: Pattern matching is a powerful tool in data-oriented programming that allows developers to extract relevant data from complex patterns and perform specific operations based on the structure and content of the data. <br><br>Ballerina provides concise and expressive pattern matching techniques to handle intricate data structures efficiently.
+description: Pattern matching is a powerful tool in data-oriented programming that allows developers to extract relevant data from complex patterns and perform specific operations based on the structure and content of the data. <br><br>Ballerina provides concise and expressive pattern-matching techniques to handle intricate data structures efficiently.
```

Comment:
No need to have hyphen https://en.wikipedia.org/wiki/Pattern_matching

Reply:
We don't need the hyphen when we are referring to the word `pattern matching` as a noun or a verb directly.
E.g., `Pattern matching is a powerful tool in data-oriented programming.`

However, we need the hyphen when it is used as an adjective or adverb that describes another noun or verb.
E.g., `Ballerina provides concise and expressive pattern-matching techniques.`
In this example, `pattern matching` acts as an adjective to describe the noun `techniques`.

Reply:
I usually see it without hyphen. Some examples.

https://www.ni.com/docs/en-US/bundle/ni-vision-concepts-help/page/pattern_matching_techniques.html

https://www.geeksforgeeks.org/applications-of-string-matching-algorithms/

https://www.javatpoint.com/pattern-matching-algorithm-in-c

Following is what chatgpt says: 

Both "pattern matching techniques" and "pattern-matching techniques" can be correct, but their usage depends on the context and style guidelines.

Pattern Matching Techniques:
In this case, "pattern matching" is treated as a noun phrase, and "techniques" is a separate noun that describes the type of methods or approaches being discussed. This usage is common when talking about various methods used for pattern matching in different domains, like computer science, linguistics, or any other field.

Pattern-Matching Techniques:
Here, "pattern-matching" is treated as a compound adjective that modifies the noun "techniques." This form is used when you want to emphasize that the techniques are specifically related to or designed for pattern matching.

In general, if the term "pattern matching" is a well-established compound concept within the context you are using it, then "pattern-matching techniques" could be more appropriate. If you're referring to various techniques related to the broader concept of pattern matching, then "pattern matching techniques" might be more suitable.

---------------

As per the last sentence of it ` If you're referring to various techniques related to the broader concept of pattern matching, then "pattern matching techniques" might be more suitable.` I feel the version without hyphen is the better one. WDYT?

Reply:
Removed the hyphen.